tests/libinstancetype/instancetype.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package libinstancetype
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+func CheckForVMInstancetypeRevisionNames(vmName string, virtClient kubecli.KubevirtClient) func() error {
+	return func() error {
+		vm, err := virtClient.VirtualMachine(util.NamespaceTestDefault).Get(vmName, &metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if vm.Spec.Instancetype.RevisionName == "" {
+			return fmt.Errorf("instancetype revision name is expected to not be empty")
+		}
+
+		if vm.Spec.Preference.RevisionName == "" {
+			return fmt.Errorf("preference revision name is expected to not be empty")
+		}
+		return nil
+	}
+}
+
+func WaitForVMInstanceTypeRevisionNames(vmName string, virtClient kubecli.KubevirtClient) {
+	Eventually(CheckForVMInstancetypeRevisionNames(vmName, virtClient), 300*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
```

Comment:
Maybe `ThisVM()` can be used here instead of getting the VM by hand?

Reply:
Possibly but I'd rather replace both `WaitForVMInstanceTypeRevisionNames` and `CheckForVMInstancetypeRevisionNames` with a matcher in a follow up if that's okay?

Reply:
Of course, a follow up makes sense.